osu.Game/Users/UserRankPanel.cs

```diff
@@ -0,0 +1,212 @@
+                                },
+                                Content = new[]
+                                {
+                                    new[]
+                                    {
+                                        Empty(),
+                                        Empty(),
+                                        Empty(),
+                                        Empty()
+                                    },
+                                    new[]
+                                    {
+                                        Empty(),
+                                        CreateAvatar().With(avatar =>
+                                        {
+                                            avatar.Size = new Vector2(60);
+                                            avatar.Masking = true;
+                                            avatar.CornerRadius = 6;
+                                        }),
+                                        new Container
+                                        {
+                                            RelativeSizeAxes = Axes.Both,
+                                            Padding = new MarginPadding { Left = padding },
+                                            Child = new GridContainer
+                                            {
+                                                RelativeSizeAxes = Axes.Both,
+                                                ColumnDimensions = new[]
+                                                {
+                                                    new Dimension()
+                                                },
+                                                RowDimensions = new[]
+                                                {
+                                                    new Dimension(GridSizeMode.AutoSize),
+                                                    new Dimension()
+                                                },
+                                                Content = new[]
+                                                {
+                                                    new Drawable[]
+                                                    {
+                                                        details = new FillFlowContainer
+                                                        {
+                                                            AutoSizeAxes = Axes.Both,
+                                                            Direction = FillDirection.Horizontal,
+                                                            Spacing = new Vector2(6),
+                                                            Children = new Drawable[]
+                                                            {
+                                                                CreateFlag(),
+                                                            }
+                                                        }
```

Comment:
i'm hoping this was just bad copy and paste and you haven't used an AI to write this.

Reply:
All of the top layer of the card was pretty much one to one copied from the `UserGridPanel`. If you're wondering why its a flow - its for the supporter icon which is being added later

Reply:
Can you add inline comments for places where code doesn't explain itself?

Reply:
Sure, should I do the same in the `UserGridCard` as well?

Reply:
Yeah, please do.